packages/js/src/server/integrations/uncaught_exception.ts

```diff
@@ -25,49 +25,59 @@ function hasOtherUncaughtExceptionListeners() {
   return process.listeners('uncaughtException').length > 1
 }
 
+function handleUncaughtException(uncaughtError: Error, client: typeof Client) {
+  if (isReporting) { 
+    return 
+  }
+
+  // In the "load" function in the exported plugin below, we 
+  // never attach this listener if enableUncaught is false...
+  // so why do we need this?
```

Comment:
Without knowing the purpose of this piece of code I am afraid to remove it. I think it might be intended for the case in which `client.config.enableUncaught` changes after the initial setup, ie the user configured Honeybadger with default config first, then ran `honeybadger.configure({ enableUncaught: false })`. It seems that [plugins will only be loaded once](https://github.com/honeybadger-io/honeybadger-js/blob/c41074ec5e78c2992ab67a171dc2a2ef91927a65/packages/core/src/client.ts#L121-L122), so that could make sense. 

Even if this is the reasoning, I still think it's a little odd to run `afterUncaught` if `enableUncaught` is false. As a user, I don't think I'd expect `afterUncaught` to run if I had disabled uncaught error handling entirely. 

Also I don't think it would work the other way... if the user set up Honeybadger with `enableUncaught = false` and later ran `honeybadger.configure({ enableUncaught: true })`, it wouldn't start reporting errors because `load()` already ran and won't run again, so no listener is attached. 

I haven't tried testing these different scenarios. Before I go down a rabbit hole, I wanted to just see if anyone else (@subzero10?) had insight.

Reply:
> I think it might be intended for the case in which client.config.enableUncaught changes after the initial setup, ie the user configured Honeybadger with default config first, then ran honeybadger.configure({ enableUncaught: false }).

Yes, I think this is the reason as well.

> Also I don't think it would work the other way... if the user set up Honeybadger with enableUncaught = false and later ran honeybadger.configure({ enableUncaught: true }), it wouldn't start reporting errors because load() already ran and won't run again, so no listener is attached.

You are correct! Plugins are loaded only once, no matter how many times you call `configure`.

Based on the above, we could:
- [my preference] change the logic to reload this plugin if `enableUncaught` is changed after plugins are loaded
- or always attach the listener and report only if `enableUncaught` is true

BUT, I would first have another look at the `uncaughtExceptionMonitor`, which: - captures both exceptions and promise rejections
- allows us to avoid checking for number of listeners and deciding whether to exit or not

Reply:
Makes sense to me! In that case let's not change anything with this PR and instead look at the `uncaughtExceptionMonitor` issue first.

Reply:
Added https://github.com/honeybadger-io/honeybadger-js/issues/1163 for this. I'll also clean up the comment in here.